coderd/workspaceapps/apptest/apptest.go

```diff
@@ -1549,3 +1564,31 @@ func testReconnectingPTY(ctx context.Context, t *testing.T, client *codersdk.Cli
 	// Ensure the connection closes.
 	require.ErrorIs(t, tr.ReadUntil(ctx, nil), io.EOF)
 }
+
+// Accessing an app should update the workspace's LastUsedAt.
+// NOTE: Despite our efforts with the flush channel, this is inherently racy.
+func assertWorkspaceLastUsedAtUpdated(t testing.TB, details *Details) {
+	t.Helper()
+
+	<-time.After(testutil.IntervalMedium) // Wait for Bicopy to finish.
+	details.FlushStats()
```

Comment:
Same here, looping flush seems better than waiting an arbitrary time?

Reply:
A loop-flush here would probably just execute after the first iteration.